### DISCUSSION TITLE
Scalar not displaying endpoints when App (.NET) is running under subpath

### DISCUSSION THREAD

**Discussion by BieleckiLtd** [scalar/scalar#3930] [Q&A] [answered]
I’m running a .NET application using Scalar for OpenAPI documentation, and the app is hosted under a subpath (/site). The OpenAPI JSON is being served correctly at https://localhost:61949/site/openapi/v1.json, and the Scalar UI is accessible at https://localhost:61949/site/scalar/v1.

However, the UI is not displaying any endpoints. It shows the UI interface but doesn’t load the API documentation properly. I suspect the issue is related to the subpath configuration. Can anyone advise please?

```
app.MapOpenApi();
app.MapScalarApiReference(options =>
{
    options.WithOpenApiRoutePattern("site/openapi/{documentName}.json");
});
```
Schema:
```
{
    "openapi": "3.0.1",
    "info": {
        "title": "MyApp | v1",
        "version": "1.0.0"
    },
    "servers": [
        {
            "url": "http://localhost:61949/SITE"
        },
        {
            "url": "https://localhost:44393/SITE"
        }
    ],
    "paths": {
        "/api/User/GetCurrentWindowsUserName": {
            "get": {
                "tags": [
                    "User"
                ],
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
    },
    "components": {},
    "tags": [
        {
            "name": "User"
        }
    ]
}
```

![image](https://github.com/user-attachments/assets/a19167f9-d790-4280-8355-1db8b1b6f21f)

**Comment by xC0dex** [comment, marked as answer]
Hey @BieleckiLtd,
thanks for the detailed explanation of the problem. I think adding a leading slash should do the trick:

```c#
app.MapScalarApiReference(options =>
{
    options.WithOpenApiRoutePattern("/site/openapi/{documentName}.json");
});
```

**Off-topic:**
I'm currently experimenting with dynamically fetching the subpath to eliminate the need to manually specify it in the configuration.

**Reply by BieleckiLtd** [reply]
Fantastic, that worked. Thanks for the quick response!
